src/blocks/frontend/live-search/index.ts

```diff
@@ -0,0 +1,387 @@
+/**
+ * Internal dependencies
+ */
+import { domReady, rgb2hsl, debounce } from '../../helpers/frontend-helper-functions.js';
+import { page, post, product, generic } from './icons';
+
+type ResultsEntry = {
+	id: number,
+	link: string,
+	title: string,
+	type: string,
+	date: string,
+	author: string,
+	parent?: string,
+	price?: string
+}
+
+type ResultsContainer = Element | null | undefined;
+
+const getPostIcon = ( type: string ) => {
+	switch ( type ) {
+	case 'post': return post();
+	case 'page': return page();
+	case 'product': return product();
+	default: return generic();
+	}
+};
+
+const getMeta = ( entry: ResultsEntry ) => {
+	const meta = document.createElement( 'div' );
+
+	switch ( entry.type ) {
+	case 'post': {
+		meta.innerHTML = `${entry.date} / by ${entry.author}${ entry.parent ? ' / ' + entry.parent : '' }`;
+		break;
+	}
+	case 'page': {
+		meta.innerHTML = entry.parent ?? '';
+		break;
+	}
+	case 'product': {
+		meta.innerHTML = entry.price ?? '';
+		break;
+	}
+	default: return null;
+	}
+
+	meta.classList.add( 'meta' );
+	return meta;
+};
+
+domReady( () => {
+	const CONTAINER_CLASS = 'search-results';
+
+	const { nonce, restUrl, strings } = window.liveSearchData;
+	const liveSearch = document.querySelectorAll( '.o-live-search' );
+	const loadingIcon = '<svg class="spinner" viewBox="0 0 100 100" width="16" height="16" xmlns="http://www.w3.org/2000/svg" focusable="false" style="width: calc(16px); height: calc(16px);"><circle cx="50" cy="50" r="50" vector-effect="non-scaling-stroke" class="main-circle"></circle><path d="m 50 0 a 50 50 0 0 1 50 50" vector-effect="non-scaling-stroke" class="moving-circle"></path></svg>';
+
+	const requestData = async( search: string, postTypes: Array<string> ) => {
+		const options = {
+			method: 'GET',
+			headers: {
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+				'x-wp-nonce': nonce
+			}
+		};
+
+		const params = postTypes.reduce( ( p, type ) => p + `&post_type[]=${type}`, `s=${search}` );
+
+		const response = await fetch( `${restUrl}?${params}`, options ).catch( error => {
```

Comment:
@arinaturcu One thing just before I was to merge, this doesn't work if you aren't using Pretty Permalinks and instead use the default, we need to check for that and replace the `?` with `&`.